### PR TITLE
Unit Tests for verifyBrandCreation

### LINKED ISSUE #71: feature/unit-testing-for-backend
## User Story: (4 points)
As a user I want to be able to use the website without any obvious bugs because that enhances the user experience

## Brief Description
Create a test suite for the backend

## Feature Requirements
-  Add Jest to backend
- Create a small test suite

# Additional context
This is going to be the precursor to automatic deployments to google cloud

### PR DESCRIPTION
**Pull request by emma-amar**
## Brief Description
created tests for the verify brand creation helper function

## Linked Issues
- closes #71 

## Changes
- tests for all cases where a brand can be created 

## Acceptance Criteria
- [x] Test coverage >80%
- [ ] All specifications met in feature requirements

### CODE REVIEW

**Review by oumbarry — APPROVED** [review]
Nice emma!

**Comment by jungsoolee1** [comment]
The "No description" test does not work on my VSCode for some reason. If others do not have the same issue as I do, then it is probably an error from my end.

**Comment by emma-amar** [comment]
> The "No description" test does not work on my VSCode for some reason. If others do not have the same issue as I do, then it is probably an error from my end. Also, it passes all the checks so it is most likely from my end.

hmmm so checks are passing when description is undefined, and we expect it to equal true (but the tests are suddenly failing on my end?) but checks are failing when we expect it to be false (and tests are passing) :-//

because description is not required, a brand should be able to be created when there isn't one. so requestBody should be true. not sure what's going on here!

**Comment by KooshaG** [comment]
> > The "No description" test does not work on my VSCode for some reason. If others do not have the same issue as I do, then it is probably an error from my end. Also, it passes all the checks so it is most likely from my end.
> 
> hmmm so checks are passing when description is undefined, and we expect it to equal true (but the tests are suddenly failing on my end?) but checks are failing when we expect it to be false (and tests are passing) :-//
> 
> because description is not required, a brand should be able to be created when there isn't one. so requestBody should be true. not sure what's going on here!

Hey Emma, I looked through the repo and it seems like you forgot to pull origin in a while so the code that you were testing still had the description check. I merged the branch with master and it seems like everything works! (at least for me lmao)

**Comment by emma-amar** [comment]
> > > The "No description" test does not work on my VSCode for some reason. If others do not have the same issue as I do, then it is probably an error from my end. Also, it passes all the checks so it is most likely from my end.
> > 
> > 
> > hmmm so checks are passing when description is undefined, and we expect it to equal true (but the tests are suddenly failing on my end?) but checks are failing when we expect it to be false (and tests are passing) :-//
> > because description is not required, a brand should be able to be created when there isn't one. so requestBody should be true. not sure what's going on here!
> 
> Hey Emma, I looked through the repo and it seems like you forgot to pull origin in a while so the code that you were testing still had the description check. I merged the branch with master and it seems like everything works! (at least for me lmao)

ahhhh!! thank you!! I literally always forget to pull lol my bad ! thanks for catching it !

File: online-shopping-website/backend/tests/verifyBrandCreation.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+describe("Required Brand Creation Params Tests", () => {
+    test("No name", () => {
+        requestBody.name = undefined;
+        requestBody.description = data[1];
+
+        expect(hasRequiredBrandCreationParams(requestBody)).toEqual(false);
+    });
+
+    test("No description", () => {
```
Comment (Adamo-O) [inline online-shopping-website/backend/tests/verifyBrandCreation.test.ts:14]: Maybe change the test name to "No description, still valid" or something
Just so it's more obvious that while there's description, the brand can still be created!

**Review by Adamo-O — COMMENTED** [review]
Looks great! Really clear and concise. Exactly what we need!

File: online-shopping-website/backend/tests/verifyBrandCreation.test.ts (hunk 1 above)
Comment (emma-amar) [reply online-shopping-website/backend/tests/verifyBrandCreation.test.ts]: done!! thanks :-))

**Review by Adamo-O — APPROVED** [review]
Looks good! Nice tests!

**Review by KooshaG — APPROVED** [review]
Nice job working with Typescript!

**Pull request merged**